src/core/tests/type_prop/segment_max.cpp

```diff
@@ -0,0 +1,310 @@
+    EXPECT_EQ(op->get_output_element_type(0), element::f32);
+    EXPECT_EQ(op->get_output_partial_shape(0), (PartialShape{Dimension::dynamic(), 12, 81}));
+    EXPECT_EQ(op->get_fill_mode(), op::FillMode::ZERO);
+}
+
+TEST_F(TypePropSegmentMaxTest, segment_ids_from_graph) {
+    const auto data = std::make_shared<Parameter>(element::f32, PartialShape{3, 12, 81});
+    const auto segment_ids_input =
+        std::make_shared<Constant>(element::i32, Shape{5}, std::vector<int32_t>{0, 1, 2, 3, 4});
+    const auto begin = Constant::create(element::i64, Shape{1}, {0});
+    const auto end = Constant::create(element::i64, Shape{1}, {3});
+    const auto stride = Constant::create(element::i64, Shape{1}, {1});
+    const auto segment_ids = std::make_shared<op::v1::StridedSlice>(segment_ids_input,
+                                                                    begin,
+                                                                    end,
+                                                                    stride,
+                                                                    std::vector<int64_t>{0},
+                                                                    std::vector<int64_t>{0});
+
+    const auto op = make_op(data, segment_ids, op::FillMode::ZERO);
+    op->validate_and_infer_types();
+    EXPECT_EQ(op->get_output_element_type(0), element::f32);
+    EXPECT_EQ(op->get_output_partial_shape(0), (PartialShape{3, 12, 81}));
+    EXPECT_EQ(op->get_fill_mode(), op::FillMode::ZERO);
+}
+
+TEST_F(TypePropSegmentMaxTest, dynamic_segment_ids_from_graph) {
+    const auto data = std::make_shared<Parameter>(element::f32, PartialShape{3, 12, 81});
+    const auto segment_ids_input = std::make_shared<Parameter>(element::i32, PartialShape::dynamic());
+    const auto begin = Constant::create(element::i64, Shape{1}, {0});
+    const auto end = Constant::create(element::i64, Shape{1}, {3});
+    const auto stride = Constant::create(element::i64, Shape{1}, {1});
+    const auto segment_ids = std::make_shared<op::v1::StridedSlice>(segment_ids_input,
+                                                                    begin,
+                                                                    end,
+                                                                    stride,
+                                                                    std::vector<int64_t>{0},
+                                                                    std::vector<int64_t>{0});
+
+    const auto op = make_op(data, segment_ids, op::FillMode::ZERO);
+    op->validate_and_infer_types();
+    EXPECT_EQ(op->get_output_element_type(0), element::f32);
+    EXPECT_EQ(op->get_output_partial_shape(0), (PartialShape{Dimension::dynamic(), 12, 81}));
+    EXPECT_EQ(op->get_fill_mode(), op::FillMode::ZERO);
+}
+
+TEST_F(TypePropSegmentMaxTest, interval_segment_ids_from_graph) {
+    const auto data = std::make_shared<Parameter>(element::f32, PartialShape{3, 12, 81});
```

Comment:
Great to see a test for such subgraph test, 
can a test case where the `num_segments` is a subgraph with ShapeOf and marked with symbol be considered too?

Reply:
`Squeeze` can be used to make a scalar from 1D tensor with single value, but allowing the input shape to be scalar or 1D with single element is also a common approach.